posts/2021-02-19-configure-trustedHeaderOrigin-21002.adoc

```diff
@@ -0,0 +1,172 @@
++
+When the entry is `/${shared.config.dir}/server.xml”` and the environment variable is `WLP_USER_DIR=/test/` in the `test/config/server.xml` file the user recieves an error stating that "it couldn't find file at /user/test/wlp/usr/shared/config/server.xml" where it should have been looking for `test/config/server.xml`.
+
+This bug caused include tags to not be parsed correctly by not replacing the environment variables found link:https://openliberty.io/docs/latest/reference/config/server-configuration-overview.html#server-env[here]
+
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/15550[NullPointerException in HttpServletRequest or HttpServletResponse context proxies]
++
+Using the `HttpServletRequest` or `HttpServletResponse` JAX-RS context proxies with a null runtime context previously may have resulted in a `NullPointerException` similar to the following:
+
+[source, java]
+----
+java.lang.NullPointerException
+at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
+at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
+at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
+at java.lang.reflect.Method.invoke(Method.java:508)
+at com.ibm.ws.jaxrs20.injection.HttpServletRequestInjectionProxy$1.invoke(HttpServletRequestInjectionProxy.java:58)
+----
+
+This issue was resolved by adding a null checks to these two proxy classes similar to those in other proxy classes.
+
+
+* link: https://github.com/OpenLiberty/open-liberty/issues/15040[ClassCastException when serving a static resource]
++
+When the following conditions occur:
+
+1. servlet 4.0
+2. The request is for a static resource
+3. The application wraps the request object in its filter.
+4. The ClassCastException might occur near the end of the request cycle when the server trying to cache the information for the served static resource.
+
+`A java.lang.ClassCastException` might occur near the end of request cycle for a static resource (i.e xml, html, jpg, css...). The response is still sending back to a client normally without any issue.
+
+An example of the exception:
+
+[source, java]
+----
+java.lang.ClassCastException: com.example.MyWrappedRequest incompatible with com.ibm.ws.webcontainer40.srt.SRTServletRequest40
+at com.ibm.ws.webcontainer40.servlet.CacheServletWrapper40.(CacheServletWrapper40.java:57)
+at com.ibm.ws.webcontainer40.servlet.factory.CacheServletWrapperFactory40Impl.createCacheServletWrapper(CacheServletWrapperFactory40Impl.java:30)
+at com.ibm.ws.webcontainer.WebContainer.addToCache(WebContainer.java:1231)
+at com.ibm.ws.webcontainer.extension.DefaultExtensionProcessor.handleRequest(DefaultExtensionProcessor.java:538)
+at com.ibm.ws.webcontainer.filter.WebAppFilterChain.invokeTarget(WebAppFilterChain.java:182)
+----
+
+This bug has been removed by fixing a `ClassCastException` when caching a static servlet wrapper under `servlet-4.0` feature.
+
```

Comment:
don't forget to add an 
`// end::features[]`  tag

Reply:
Done